File: Python Programs/Linear_search.py
#####################################
# LINEAR SEARCH
#####################################

#getting array input with integer elements.
array = input("Enter the elements(Use space to seperate):").split()

#value to find
value = input("Enter the value you want to search:")

#Note that both the array and value are holding string datatypes.

#Linear search
print("Value:"+value+" is at,")
validator = 0  #to check whether the element exists or not
for index in range(len(array)):
    if array[index] == value:
        validator = 1
        print("Index:",index)
if (validator == 0):   print("No index found")
        
        

File: C Programs/Linear_search.c
#include <stdio.h>

// This is a linear search program.

int main()
{   int size;
    int value;
    int validator = 0;  //validator is to check whether the elements to find exists or not
    
    // getting the array input
    printf("Enter the size of the array:");
    scanf("%d",&size);
    int array[size];
    for(int i = 0;i<size;i++){
        printf("Enter the element no %d:",i+1);
        scanf("%d",&array[i]);
    }
    
    // value to be find
    printf("Enter the element you want to search:");
    scanf("%d",&value);
    
    //linear search
    printf("The value:- %d is at,\n",value);
    
    for(int i = 0;i<size;i++){
        if (array[i] == value){
            validator = 1;
            printf("index:%d \n",i);
        }
    }
    if(validator == 0) {printf("This element is not in the array");}
    
    return 0;
}
